pycbc/transforms.py

```diff
@@ -332,22 +332,102 @@ def inverse_transform(self, maps):
         return self.format_output(maps, out)
 
     def jacobian(self, maps):
-        """Returns the Jacobian for the transforming mchirp and q to mass1 and
+        """Returns the Jacobian for transforming mchirp and q to mass1 and
         mass2.
         """
         mchirp = maps['mchirp']
         q = maps['q']
         return mchirp * ((1.+q)/q**3.)**(2./5)
 
     def inverse_jacobian(self, maps):
-        """Returns the Jacobian for the transforming mass1 and mass2 to
+        """Returns the Jacobian for transforming mass1 and mass2 to
         mchirp and q.
         """
         m1 = conversions.primary_mass(maps['mass1'], maps['mass2'])
         m2 = conversions.secondary_mass(maps['mass1'], maps['mass2'])
         return conversions.mchirp_from_mass1_mass2(m1, m2)/m2**2.
 
 
+class ChirpDistanceToDistance(BaseTransform):
+    """ Converts chirp distance to luminosity distance, given the chirp mass.
+    """
+    name = "chirp_distance_to_distance"
+    _inputs = [parameters.chirp_distance]
+    _outputs = [parameters.distance, parameters.mchirp]
+
+
+    def transform(self, maps):
+        """This function transforms from luminosity distance to chirp distance,
+        given the chirp mass.
+        Parameters
+        ----------
+        maps : a mapping object
+        Examples
+        --------
+        Convert a dict of numpy.array:
+        >>> import numpy as np
+        >>> from pycbc import transforms
+        >>> t = transforms.ChirpDistanceToDistance()
+        >>> t.transform({'chirp_distance': np.array([40.]), 'mchirp': np.array([1.2])})
+        {'mchirp': array([ 1.2]), 'chirp_distance': array([ 40.]), 'distance': array([ 39.48595679])}
+        Returns
+        -------
+        out : dict
+            A dict with key as parameter name and value as numpy.array or float
+            of transformed values.
+        """
+        out = {}
+        out[parameters.distance] = \
+                 conversions.distance_from_chirp_distance_mchirp(maps[parameters.chirp_distance],
+                                                     maps[parameters.mchirp])
```

Comment:
Pass the reference mass here; see my comment below about storing that as an instance attribute.